prbt_support/launch/robot.launch

```diff
@@ -36,57 +17,76 @@
-  <node ns="prbt" name="controller_spawner" pkg="controller_manager" type="spawner"
-        args="$(arg spawn_controllers)" />
-
-  <!-- Initialize controllers and start STO-processing (if configured) -->
-  <group unless="$(eval sto.lower() == 'none')" >
-    <!-- Load api spec for all nodes -->
-    <rosparam ns="/prbt/read_api_spec" command="load" file="$(arg read_api_spec_file)" />
-    <rosparam ns="/prbt/write_api_spec" command="load" file="$(arg write_api_spec_file)" if="$(arg has_operation_mode_support)" />
-
-    <node pkg="rosservice" type="rosservice" name="robot_init" args="call --wait /prbt/driver/init"/>
-    <include file="$(find prbt_hardware_support)/launch/modbus_client.launch">
-      <arg name="modbus_server_ip" value="$(arg sto_modbus_server_ip)" />
-    </include>
-    <include file="$(find prbt_hardware_support)/launch/modbus_adapter_sto_node.launch" />
-    <include file="$(find prbt_hardware_support)/launch/stop1_executor_node.launch" />
-    <include if="$(arg has_braketest_support)" file="$(find prbt_hardware_support)/launch/brake_test.launch">
-      <arg name="write_api_spec_file" value="$(arg write_api_spec_file)" />
-    </include>
-
-    <include if="$(arg has_operation_mode_support)" file="$(find prbt_hardware_support)/launch/operation_mode.launch" />
-    <group unless="$(arg has_operation_mode_support)">
-      <node name="fake_speed_override_node" pkg="prbt_hardware_support" type="fake_speed_override_node"/>
-    </group>
+  <!-- Spawn controllers -->
+  <node ns="prbt" name="controller_spawner" pkg="controller_manager" type="spawner" args="$(arg spawn_controllers)" />
+
+  <!-- Initialize driver -->
+  <node pkg="rosservice" type="rosservice" name="robot_init" args="call --wait /prbt/driver/init"/>
+
+  <!-- Read modbus register specifications -->
+  <rosparam ns="/prbt/read_api_spec" command="load" file="$(arg read_api_spec_file)" />
+  <rosparam ns="/prbt/write_api_spec" command="load" file="$(arg write_api_spec_file)" if="$(arg has_operation_mode_support)" />
+
+  <!-- Open modbus connection and required modules -->
+  <include file="$(find prbt_hardware_support)/launch/modbus_client.launch">
+    <arg name="modbus_server_ip" value="$(arg modbus_server_ip)" />
+  </include>
+  <include file="$(find prbt_hardware_support)/launch/modbus_adapter_run_permitted_node.launch" />
+  <include file="$(find prbt_hardware_support)/launch/stop1_executor_node.launch" />
+  <include if="$(arg has_braketest_support)" file="$(find prbt_hardware_support)/launch/brake_test.launch">
+    <arg name="write_api_spec_file" value="$(arg write_api_spec_file)" />
+  </include>
+  <include if="$(arg has_operation_mode_support)" file="$(find prbt_hardware_support)/launch/operation_mode.launch" />
+  <group unless="$(arg has_operation_mode_support)">
+    <node name="fake_speed_override_node" pkg="prbt_hardware_support" type="fake_speed_override_node"/>
   </group>
 
-  <group if="$(eval sto.lower() == 'none')" >
```

Comment:
Why did you drop the init?

Reply:
We have decided, not to support this as an option any more. Users can still do this if they want, but it is not supported by us.

Reply:
How can I startup without modbus related nodes without forking pilz_robots?